### ISSUE TITLE
set, add, remove and delete operations set twice or more

### ISSUE THREAD

**Issue by Freezystem** [brabster/dynamodb-expressions#12]
I'm using a `reduce()` to dynamically generate the correct update expression with your helper. 
I want to update my row like this:
 - remove non existing fields
 - set existing ones.

Looks like:
```clj
(defn generate-update-expression
  [id m]
  (let [fields [:field1 :field2 :field3 :field4]]
    (-> (reduce (fn [ue f] (if-let [v (f m)] (dx/set ue f v) (dx/remove ue f)))
                (dx/update-expr {:id id}) fields)
        dx/expr)))
```

As I don't know which ones are present or not, values are added as they come, giving something like: 
```clj
(-> (dx/update-expr {:id id})
    (dx/set :field1 (:field1 m))
    (dx/set :field2 (:field2 m))
    (dx/remove :field3)
    (dx/set :field4 (:field4 m))
    dx/expr)
```

Which generate the following invalid dynamoDB update expression:
`SET #nfield1_15365 = :vfield1_15365, #nfield2_15366 = :vfield2_15366 REMOVE #nfield3_15368 SET #nfield4_15367 = :vfield4_15367`

Is this an expected behavior ? AFAIK I don't think it is documented.

**Comment by brabster** [comment]
This problem should be fixed in release 0.4.0
https://github.com/brabster/dynamodb-expressions/releases/tag/0.4.0

**Comment by Freezystem** [comment]
Thanks a lot for you time, it's working great.
